libs/wingsdk/src/target-sim/bucket.inflight.ts

```diff
@@ -250,15 +256,37 @@ export class Bucket implements IBucketClient, ISimulatorResourceInstance {
     });
   }
 
-  private async addFile(key: string, value: string): Promise<void> {
+  private async storeMetadata(
+    filePath: string,
+    contentType: string
+  ): Promise<void> {
+    const metadataPath = filePath + ".meta";
+    const metadata = {
+      contentType: contentType,
+    };
+    fs.writeFileSync(metadataPath, JSON.stringify(metadata));
+  }
+
+  private async addFile(
+    key: string,
+    value: string,
+    contentType?: string
+  ): Promise<void> {
     const actionType: BucketEventType = this.objectKeys.has(key)
       ? BucketEventType.UPDATE
       : BucketEventType.CREATE;
+
     const hash = this.hashKey(key);
     const filename = join(this._fileDir, hash);
     const dirName = dirname(filename);
+
     await fs.promises.mkdir(dirName, { recursive: true });
     await fs.promises.writeFile(filename, value);
+
+    const determinedContentType =
+      contentType ?? mime.getType(key) ?? "application/octet-stream";
+    await this.storeMetadata(filename, determinedContentType);
```

Comment:
Couldn't we store this in a simple `Map` or `Record<string, Metadata>`? The metadata is pretty small and it will probably be faster as well
```suggestion
    this.metadata[key] = { contentType: determinedContentType };
```

Reply:
`metadata` is a method, so how can you assign a value to it?